From f4d3e9fd16ddbeb78f38b8aeb8b3dcc55897186a Mon Sep 17 00:00:00 2001
From: Wallace Wang <r59996@freescale.com>
Date: Tue, 8 Jul 2008 17:46:13 +0800
Subject: [PATCH] ENGR00083285 MX35: ESAI buffer is disturbed by zero filling

1. DMA is enabled first, the data has been transfered to ESAI fifo
2. ESAI driver fill some zero data to initialize the TX reg.
3. The filled zero data overwrite the ESAI FIFO

Signed-off-by: Wallace Wang <r59996@freescale.com>
---
 sound/soc/imx/imx-esai.c |    7 ++-----
 1 files changed, 2 insertions(+), 5 deletions(-)

diff --git a/sound/soc/imx/imx-esai.c b/sound/soc/imx/imx-esai.c
index 8b85255..7111445 100644
--- a/sound/soc/imx/imx-esai.c
+++ b/sound/soc/imx/imx-esai.c
@@ -653,7 +653,7 @@ static int imx_esai_hw_tx_params(struct snd_pcm_substream
 	channels = params_channels(params);
 	tfcr &= ESAI_TFCR_TE_MASK;
 	tfcr |= ESAI_TFCR_TE(channels);
-	tfcr |= ESAI_TFCR_TFEN | ESAI_TFCR_TIEN;
+	tfcr |= ESAI_TFCR_TFEN;
 
 	tfcr |= ESAI_TFCR_TFWM(64);
 
@@ -734,7 +734,7 @@ static int imx_esai_hw_params(struct snd_pcm_substream
 
 static int imx_esai_trigger(struct snd_pcm_substream *substream, int cmd)
 {
-	u32 i, reg;
+	u32 reg;
 
 	if (substream->stream == SNDRV_PCM_STREAM_PLAYBACK) {
 		reg = __raw_readl(ESAI_TCR);
@@ -747,9 +747,6 @@ static int imx_esai_trigger(struct snd_pcm_substream *substream, int cmd)
 	}
 	switch (cmd) {
 	case SNDRV_PCM_TRIGGER_START:
-		/* Before SDMA data transfer, fifo needs to be initialized */
-		for (i = 0; i < 120; i++)
-			__raw_writel(0x00, ESAI_ETDR);
 	case SNDRV_PCM_TRIGGER_RESUME:
 	case SNDRV_PCM_TRIGGER_PAUSE_RELEASE:
 		if (substream->stream == SNDRV_PCM_STREAM_PLAYBACK)
-- 
1.5.4.4

